src/map/clif.c

```diff
@@ -20377,6 +20377,23 @@ static void clif_monster_hp_bar(struct mob_data *md, struct map_session_data *sd
 #endif
 }
 
+static bool clif_show_monster_hp_bar(struct block_list *bl) {
+
+	nullpo_ret(bl);
+
+	const struct mob_data *md = BL_UCCAST(BL_MOB, bl);
+
+	if (bl->type == BL_MOB && status_get_hp(bl) < status_get_max_hp(bl)) {
+		if ((battle->bc->show_monster_hp_bar & 1 && !(md->class_ == MOBID_EMPELIUM) && !(md->status.mode & MD_BOSS))
```

Comment:
here checks look like wrong.
use ``()`` for separate blocks with ``&``
for example like this ``if (((battle->bc->show_monster_hp_bar & 1) && !(md->class_ == MOBID_EMPELIUM) ...``
same for ``& 2`` and ``& 4``

Reply:
Please see changes in latest commit